api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -186,115 +183,97 @@
-    try {
-      fcCluster = this.leonardoNotebooksClient.getCluster(billingProjectId, clusterName);
-    } catch (NotFoundException e) {
-      fcCluster =
-          this.leonardoNotebooksClient.createCluster(
-              billingProjectId, clusterName, workspaceFirecloudName);
-    }
+  @Override
+  public ResponseEntity<Cluster> createCluster(String workspaceNamespace) {
+    String firecloudWorkspaceName = lookupWorkspace(workspaceNamespace).getFirecloudName();
+    workspaceService.enforceWorkspaceAccessLevel(
+        workspaceNamespace, firecloudWorkspaceName, WorkspaceAccessLevel.WRITER);
+    workspaceService.validateActiveBilling(workspaceNamespace, firecloudWorkspaceName);
 
-    int retries = Optional.ofNullable(user.getClusterCreateRetries()).orElse(0);
-    if (org.pmiops.workbench.notebooks.model.ClusterStatus.ERROR.equals(fcCluster.getStatus())) {
-      if (retries <= 2) {
-        this.userService.setClusterRetryCount(retries + 1);
-        log.warning("Cluster has errored with logs: ");
-        if (fcCluster.getErrors() != null) {
-          for (ClusterError e : fcCluster.getErrors()) {
-            log.warning(e.getErrorMessage());
-          }
-        }
-        log.warning("Retrying cluster creation.");
-
-        this.leonardoNotebooksClient.deleteCluster(billingProjectId, clusterName);
-      }
-    } else if (org.pmiops.workbench.notebooks.model.ClusterStatus.RUNNING.equals(
-            fcCluster.getStatus())
-        && retries != 0) {
-      this.userService.setClusterRetryCount(0);
-    }
-    DefaultClusterResponse resp = new DefaultClusterResponse();
-    resp.setDefaultCluster(TO_ALL_OF_US_CLUSTER.apply(fcCluster));
-    return ResponseEntity.ok(resp);
+    org.pmiops.workbench.notebooks.model.Cluster firecloudCluster =
+        leonardoNotebooksClient.createCluster(
+            workspaceNamespace, clusterNameForUser(userProvider.get()), firecloudWorkspaceName);
+
+    return ResponseEntity.ok(TO_ALL_OF_US_CLUSTER.apply(firecloudCluster));
   }
 
   @Override
-  public ResponseEntity<EmptyResponse> deleteCluster(String projectName, String clusterName) {
-    this.userService.setClusterRetryCount(0);
-    this.leonardoNotebooksClient.deleteCluster(projectName, clusterName);
+  public ResponseEntity<EmptyResponse> deleteCluster(String workspaceNamespace) {
+    String firecloudWorkspaceName = lookupWorkspace(workspaceNamespace).getFirecloudName();
+    workspaceService.enforceWorkspaceAccessLevel(
+        workspaceNamespace, firecloudWorkspaceName, WorkspaceAccessLevel.WRITER);
+
+    this.leonardoNotebooksClient.deleteCluster(
+        workspaceNamespace, clusterNameForUser(userProvider.get()));
     return ResponseEntity.ok(new EmptyResponse());
   }
 
   @Override
   public ResponseEntity<ClusterLocalizeResponse> localize(
-      String projectName, String clusterName, ClusterLocalizeRequest body) {
+      String workspaceNamespace, ClusterLocalizeRequest body) {
```

Comment:
I don't really know what `localize` means in this context, but I can see this method does a large number of things that are all important (although transaction management between service calls is potentially sloppy). It would be helpful for readability and testability to have a high-level comment and also break down the body into helper methods and service methods. (Can certainly come later)